src/script/lua_api/l_areastore.cpp

```diff
@@ -110,6 +110,9 @@ int LuaAreaStore::l_get_area(lua_State *L)
 
 	const Area *res;
 
+	if (!ast->getArea(id))
+		return 0;
+
 	res = ast->getArea(id);
```

Comment:
Well I force pushed that change, anyway, though I'm still compiling on this old laptop to test.